Assets/Tests/InputSystem/CoreTests_Actions.cs

```diff
@@ -1730,1484 +1849,996 @@ public void Actions_PressingAndReleasingButtonInSameUpdate_StillTriggersAction()
 
-            Assert.That(trace, Is.Empty);
-        }
+        // Lookup is case-insensitive.
+        Assert.That(map.TryGetAction("Action1"), Is.SameAs(action1));
+        Assert.That(map.TryGetAction("Action2"), Is.SameAs(action2));
     }
 
     [Test]
     [Category("Actions")]
-    [Property("TimesliceEvents", "Off")]
-    public void Actions_CanPerformHoldInteraction()
+    public void Actions_CanLookUpActionsInMapById()
     {
-        var timeOffset = 123;
-        runtime.currentTimeOffsetToRealtimeSinceStartup = timeOffset;
-        runtime.currentTime = 10 + timeOffset;
-        var gamepad = InputSystem.AddDevice<Gamepad>();
+        var map = new InputActionMap();
 
-        var performedReceivedCalls = 0;
-        InputAction performedAction = null;
-        InputControl performedControl = null;
+        var action1 = map.AddAction("action1");
+        var action2 = map.AddAction("action2");
 
-        var startedReceivedCalls = 0;
-        InputAction startedAction = null;
-        InputControl startedControl = null;
+        Assert.That(map.TryGetAction(action1.id), Is.SameAs(action1));
+        Assert.That(map.TryGetAction(action2.id), Is.SameAs(action2));
+        Assert.That(map.TryGetAction(Guid.NewGuid()), Is.Null);
+    }
 
-        var canceledReceivedCalls = 0;
-        InputAction canceledAction = null;
-        InputControl canceledControl = null;
+    [Test]
+    [Category("Actions")]
+    public void Actions_CanLookUpActionsInMapByStringId()
+    {
+        var map = new InputActionMap();
 
-        var action = new InputAction(binding: "<Gamepad>/{primaryAction}", interactions: "hold(duration=0.4)");
-        action.performed +=
-            ctx =>
-        {
-            ++performedReceivedCalls;
-            performedAction = ctx.action;
-            performedControl = ctx.control;
+        var action1 = map.AddAction("action1");
+        var action2 = map.AddAction("action2");
 
-            Assert.That(action.phase, Is.EqualTo(InputActionPhase.Performed));
-        };
-        action.started +=
-            ctx =>
-        {
-            ++startedReceivedCalls;
-            startedAction = ctx.action;
-            startedControl = ctx.control;
+        Assert.That(map.TryGetAction(action1.id.ToString()), Is.SameAs(action1));
+        Assert.That(map.TryGetAction(action2.id.ToString()), Is.SameAs(action2));
+        Assert.That(map.TryGetAction(Guid.NewGuid().ToString()), Is.Null);
+    }
 
-            Assert.That(action.phase, Is.EqualTo(InputActionPhase.Started));
-        };
-        action.canceled +=
-            ctx =>
-        {
-            ++canceledReceivedCalls;
-            canceledAction = ctx.action;
-            canceledControl = ctx.control;
+    // We used to require string GUIDs to be using a "{...}" format when looking up actions. We no
+    // longer do this but there's still data that may be using this format so make sure it works for now.
+    [Test]
```

Comment:
I think we discussed this before, but do we want to carry this into 1.0?

Reply:
Think we should axe all of those.

Reply:
+1